commons/src/main/java/org/infinispan/commons/util/IntSets.java

```diff
@@ -0,0 +1,195 @@
+package org.infinispan.commons.util;
+
+import java.util.PrimitiveIterator;
+import java.util.Set;
+import java.util.function.IntConsumer;
+
+/**
+ * Static utility class for creating various {@link IntSet} objects.
+ * @author wburns
+ * @since 9.3
+ */
+public class IntSets {
+   private IntSets() { }
+
+   /**
+    * Returns an immutable IntSet containing no values
+    * @return IntSet with no values
+    */
+   public static IntSet immutableEmptySet() {
+      return EmptyIntSet.getInstance();
+   }
+
+   /**
+    * Returns an immutable IntSet containing a single value
+    * @param value the value to be set on the IntSet
+    * @return IntSet with just the 1 value
+    */
+   public static IntSet immutableSet(int value) {
+      return new SingletonIntSet(value);
+   }
+
+   /**
+    * Returns an immutable IntSet that wraps the given IntSet to prevent modifications.
+    * @param set set to wrap
+    * @return immutable IntSet
+    */
+   public static IntSet immutableSet(IntSet set) {
+      return new ImmutableIntSet(set);
+   }
+
+   /**
+    * Returns an immutable IntSet containing all values from {@code 0} to {@code endExclusive - 1}.
+    * @param endExclusive max value less 1 to use
+    * @return IntSet with the values in the given range available
+    */
+   public static IntSet immutableRangeSet(int endExclusive) {
+      if (endExclusive == 0) {
+         return EmptyIntSet.getInstance();
+      }
+      if (endExclusive == 1) {
+         return new SingletonIntSet(0);
```

Comment:
`RangeSet` can't be used in this case. Unless we changed `RangeSet` to take a beginning and and end. Also I found that both are used quite a bit actually.

Reply:
Sorry I didn't realize what you meant, yeah I can see removing the `1` case. The 0 case requires no allocation though, so I would think just in case :) The 0 case is used with REPL btw.

Reply:
IMO the one usage is not a great argument: it saves an object instance per cache, before it has any state, but the extra implementation potentially affects the inlining of all the other usages.

Reply:
Okay